childcare/entity.py

```diff
@@ -0,0 +1,72 @@
+from common.constants import entity_key
+from entity.abstract import ResourceEntity
+from datetime import datetime, timedelta
+
+import json
+import requests
+
+API_URL = 'https://data.cityofnewyork.us/resource/dsg6-ifza.json' + \
+    '?$select=`centername`,`legalname`,`zipcode`,`status`,`dc_id`,`childcaretype`' + \
+    '&$where=inspectiondate >= \'{}\' and inspectiondate < \'{}\'&$limit=10000&$offset={}' + \
+    '&$order=inspectiondate'
+
+class ChildCareCenter(ResourceEntity):
+
+    @staticmethod
+    def dependencies():
+        return[
+            entity_key.census_us_zipcode,
+            entity_key.childcare_center_type
+        ]
+
+    def get_zipcode_id(self, record, field):
+        zipcode_entity = self.dependencies_map[entity_key.census_us_zipcode]
+        zipcode = zipcode_entity.get_cached_value(record[field])
+        return zipcode['id'] if zipcode else None
+
+    def get_caretype_id(self, record, field):
+        childcare_type_entity = self.dependencies_map[entity_key.childcare_center_type]
+        return childcare_type_entity.get_cached_value(record[field])['id']
+    
+    def __init__(self):
+        super().__init__()
+
+        self.table_name = 'childcare_center'
+        self.fields = [
+            {'field': 'centername', 'column': 'center_name'},
+            {'field': 'legalname', 'column': 'legal_name'},
+            {'field': 'zipcode', 'column': 'zipcode_id', 'data': self.get_zipcode_id},
+            {'field': 'status'},
+            {'field': 'dc_id', 'column': 'public_id'},
+            {'field': 'childcaretype', 'column': 'childcare_type_id', 'data': self.get_caretype_id},
+        ]
+        self.cacheable_fields = ['public_id']
+
+    def skip_record(self, record):
+        return self.record_cache and record['dc_id'] in self.record_cache
```

Comment:
`dc_id` doesn't seem to be unique per childcare center we will probably need to use the center_name column to identify a unique center. We will also need to put a unique index on that column as well, there are a lot of duplicates in the database.